components/canvas/webgl_thread.rs

```diff
@@ -1200,6 +1202,19 @@ impl WebGLImpl {
         chan.send(gl.get_string(gl::EXTENSIONS)).unwrap();
     }
 
+    // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.6
+    fn get_framebuffer_attachment_parameter(
+        gl: &gl::Gl,
+        target: u32,
+        attachment: u32,
+        pname: u32,
+        chan: WebGLSender<WebGLResult<WebGLParameter>>
```

Comment:
So, there's no reason this chan returns a `WebGLResult` now, right? I'd make it just return an integer, and remove the dead code from the `handle_potential_webgl_error` stuff.

Reply:
The channels require a type parameter and we can't pass `WebGLSender` to `get_framebuffer_attachment_parameter()`. We can simplify that to `WebGLSender<WebGLParameter>` and the code below would become
```Rust
match receiver.recv().unwrap() {
    WebGLParameter::Int(value) => Int32Value(value),
    _ => panic!("Framebuffer attachment parameter must be an integer"),
}
```
Sounds good?

Reply:
Or, we could do `WebGLSender<i32>`, and then `Int32Value(receiver.recv().unwrap())` would be possible. What do you think?

Reply:
Yes, if it will always return an integer then `WebGLSender<i32>` makes sense.